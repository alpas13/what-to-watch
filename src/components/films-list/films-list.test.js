import React from "react";
import renderer from "react-test-renderer";
import FilmsList from "./films-list.jsx";
import SmallMovieCard from "../small-movie-card/small-movie-card";

const films = [
  {
    id: 1,
    name: `Shutter Island`,
    posterImage: `img/shutter-island-poster.jpg`,
    previewImage: `img/shutter-island.jpg`,
    backgroundImage: `img/shutter-island-bg.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://some-link`,
    previewVideoLink: `https://some-link`,
    description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer who escaped from a hospital for the criminally insane.`,
    rating: 8.1,
    scoresCount: 610,
    director: `Martin Scorsese`,
    starring: [`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`],
    runTime: 138,
    genre: `Mystery`,
    released: 2010,
    isFavorite: false
  }
];

const film = {
  id: 1,
  name: `Shutter Island`,
  posterImage: `img/shutter-island-poster.jpg`,
  previewImage: `img/shutter-island.jpg`,
  backgroundImage: `img/shutter-island-bg.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://some-link`,
  previewVideoLink: `https://some-link`,
  description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer who escaped from a hospital for the criminally insane.`,
  rating: 8.1,
  scoresCount: 610,
  director: `Martin Scorsese`,
  starring: [`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`],
  runTime: 138,
  genre: `Mystery`,
  released: 2010,
  isFavorite: false
};

test(`Correctly render FilmsList component`, () => {
  const tree = renderer.create(
      <FilmsList
          films={films}
      >
        <SmallMovieCard
            film={film}
        />
      </FilmsList>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
