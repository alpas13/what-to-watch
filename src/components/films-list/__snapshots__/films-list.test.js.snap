// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly render FilmsList component 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Shutter Island"
        height="175"
        src="img/shutter-island.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="https://some-link"
      >
        Shutter Island
      </a>
    </h3>
  </article>
</div>
`;
